core/src/main/java/org/commonjava/maven/ext/core/groovy/InvocationStage.java

```diff
@@ -22,15 +22,17 @@
  */
 public enum InvocationStage
 {
-    FIRST(1),
-    LAST(99),
-    BOTH(Integer.MAX_VALUE);
+    PREPARSE( 0 ),
+    FIRST( 1 ),
+    LAST( 99 ),
+    BOTH( Integer.MAX_VALUE ),
+    ALL( Integer.MAX_VALUE );
```

Comment:
So I made `ALL` with the same value as `BOTH` but I am not sure it is useful as we don't want to normally run a `BOTH` script during preparse anyway.

Reply:
We should look at a way to deprecate usage of BOTH - detect and warn?

Reply:
Now, I wonder if `ALL` even makes sense since we don't want to run `PREPARSE` (which I would think would be included in `ALL`). Really, I think 'BOTH' meaning `FIRST` and `LAST` is OK, then.

Reply:
To be honest I think we have to be careful not to make assumptions on when people might want to run it. Hence wanting to deprecate BOTH and replace with ALL.

Reply:
I may be missing something.

If we make `BOTH` equivalent to `ALL` and don't change the semantics, then the name is actually misleading.

If we change the semantics, then scripts which currently work with `BOTH` are likely to break with `PREPARSE` where (`ALL = { PREPASE, FIRST, LAST }`) since it's missing, e.g., the project information.

Yet, we currently have no other way of supporting any other combinations of stages I thought.

Reply:
I know the `BOTH` name is misleading if we make it equivalent to `ALL`. Equally though its misleading now there are  _three_ states and not two. A user can still check what stage the script is in before choosing to continue processing or not. 
I do not believe `BOTH` is appropriate any more ; a more radical solution is simply to remove it but if you were to choose to do that I would suggest checking with users first?

Reply:
I have tried to keep `BOTH` in the `if` in `BaseGroovyManipulator` (by checking `FIRST` and `LAST`) if it looks right to you. In which case, I am not sure it really needs to be deprecated, we'd basically be keeping the original meaning while adding `ALL` to cover the third state (or beyond).

Reply:
I'm honestly not sure of the benefit to keeping `BOTH` as a differentiator from `ALL` ; plus it will make your upcoming documentation modifications more complex.

Reply:
We have decided to remove `BOTH`. However, I will need to check as `BOTH` is used in one of the IT setup tests shared by some other tests.

Reply:
Removed `BOTH`, changed existing IT to use `ALL` instead.